contentcuration/contentcuration/serializers.py

```diff
@@ -23,42 +23,51 @@ class Meta:
         model = Language
         fields = ('lang_code', 'lang_subcode', 'id')
 
-class UserSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = User
-        fields = ('email', 'first_name', 'last_name', 'is_active', 'is_admin', 'id','clipboard_tree')
 
-class ChannelSerializer(serializers.ModelSerializer):
-    resource_count = serializers.SerializerMethodField('count_resources')
-    resource_size = serializers.SerializerMethodField('calculate_resources_size')
-    def count_resources(self, channel):
-        if not channel.main_tree:
-            return 0
-        else:
-            return count_files(channel.main_tree)
-
-    def calculate_resources_size(self, channel):
-        if not channel.main_tree:
-            return 0
-        else:
-            return get_total_size(channel.main_tree)
+class FileListSerializer(serializers.ListSerializer):
+    def update(self, instance, validated_data):
+        file_mapping = {file_obj.id: file_obj for file_obj in instance}
+        node_file_mapping = []
+        ret = []
+        update_files = {}
 
-    class Meta:
-        model = Channel
-        fields = ('id', 'name', 'description', 'editors', 'main_tree',
-                    'clipboard_tree', 'trash_tree','resource_count', 'resource_size',
-                    'version', 'thumbnail', 'deleted', 'public', 'pending_editors')
+        with transaction.atomic():
+            for item in validated_data:
+                if 'id' in item:
+                    update_files[item['id']] = item
+                else:
+                    # create new nodes
+                    ret.append(File.objects.create(**item))
+
+        for file_obj in validated_data:
```

Comment:
Could potentially cause problems if another user is building a tree- might be better to add boolean on files to indicate whether or not they should be deleted on cleanup